java/arcs/core/host/ArcHostContext.kt

```diff
@@ -15,15 +15,20 @@ import arcs.core.common.toArcId
 import arcs.core.data.Plan
 import arcs.core.entity.Handle
 import arcs.core.host.api.Particle
+import arcs.core.storage.StorageProxy.StorageEvent
 import arcs.core.util.TaggedLog
 
 /**
  * Holds per-[Particle] context state needed by [ArcHost] to implement [Particle] lifecycle.
+ * TODO: this now has active logic and probably shouldn't be a data class
  *
  * @property particle currently instantiated [Particle] class
- * @property handles handles a map of each handle created for this [Particle]
+ * @property planParticle the [Plan.Particle] used to instantiate [particle]
+ * @property handles a map of each handle created for [particle]
  * @property particleState the current state the particle lifecycle is in
  * @property consecutiveFailureCount how many times this particle failed to start in a row
+ * @property awaitingReady set of handles that expect to receive a `ready` StorageEvent
+ * @property desyncedHandles set of handles that are currently desynchronized from storage
  */
 data class ParticleContext(
```

Comment:
This can be made a non-data class in a followup by overriding equals/hashcode/toString/copy if desired.

Reply:
Yes, I thought that would be the case, but didn't want to bloat this CL with that. I think only `copy` is actually used, so shouldn't be too hard.